### ISSUE TITLE
react/prop-types reports properties already declared through React.FC<Props>

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1224]
## Summary

Reproduced with @utoo/lint 0.4.1 on Node.js 22.23.1, macOS arm64. The component already declares its props using React.FC<{ value: string }>. The property access should not require a separate runtime propTypes declaration.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 eslint-plugin-react@7.37.5
```

`example.tsx`:

```tsx
import React from 'react';
export const Example: React.FC<{ value: string }> = (props) => {
  return <span>{props.value}</span>;
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "error"
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Result

ESLint reports no diagnostics. utlint reports one react/prop-types error for value.

Both tools check the same file successfully; there are no syntax errors or lint-disable comments.

**Comment by fireairforce** [comment]
Additional minimal reproductions on @utoo/lint 0.4.1 with the same ESLint/parser/react-plugin versions as the original report. This also affects props typed directly on a function parameter, not only React.FC annotations. Use the original rule configuration and replace example.tsx with either case below.

`example.tsx`:

```tsx
export function Example(props: {value:string}) { const {value}=props;return <span>{value}</span>; }
```

```text
ESLint:
(no diagnostics)

utlint:
'value' is missing in props validation
```

`example.tsx`:

```tsx
interface Props {value:string;} export function Example(props: Props){ return <span>{props.value}</span>; }
```

```text
ESLint:
(no diagnostics)

utlint:
'value' is missing in props validation
```

**Comment by fireairforce** [comment]
Retested with @utoo/lint 0.4.2, ESLint 8.57.1, @typescript-eslint/parser 6.21.0 and eslint-plugin-react 7.37.5 on Node.js 22.23.1 (macOS arm64).

The original direct React.FC prop access, inline parameter types, and local interface parameter cases are fixed. These additional TypeScript cases still produce false positives. Use the original setup with @utoo/lint@0.4.2 and enable only `react/prop-types`. Each example is separate.

### typed-string-method

`example.tsx`:

```tsx
import React from 'react'; export const Example:React.FC<{value:string}> = ({value}) => <span>{value.slice(1)}</span>;
```

ESLint: no diagnostics.

utlint: `'value.slice' is missing in props validation`.

### typed-array-member

`example.tsx`:

```tsx
export function Example({items}:{items:string[]}){return <span>{items.length}{items.map(x=>x)}</span>;}
```

ESLint: no diagnostics.

utlint: `'items.length' is missing in props validation`; `'items.map' is missing in props validation`.

### typed-nested-local

`example.tsx`:

```tsx
interface Data {name:string;} export function Example({data}:{data:Data}){return <span>{data.name}</span>;}
```

ESLint: no diagnostics.

utlint: no diagnostics (control).

### typed-nested-import

`example.tsx`:

```tsx
import type {Data} from './types'; export function Example({data}:{data:Data}){return <span>{data.name}</span>;}
```

`types.ts`:

```ts
export interface Data {name:string;}
```

ESLint: no diagnostics.

utlint: `'data.name' is missing in props validation`.

### typed-intersection-local

`example.tsx`:

```tsx
interface Base {value:string;} export function Example(props:Base & {other:number}){const{value}=props;return <span>{value}</span>;}
```

ESLint: no diagnostics.

utlint: no diagnostics (control).

### typed-intersection-import

`example.tsx`:

```tsx
import type {Base} from './types'; export function Example(props:Base & {other:number}){const{value}=props;return <span>{value}</span>;}
```

`types.ts`:

```ts
export interface Base {value:string;}
```

ESLint: no diagnostics.

utlint: `'value' is missing in props validation`.

The local-interface cases are controls: they demonstrate that the nested access and intersection are accepted when their types are declared locally. Imported types still trigger missing-validation diagnostics. String and array members also remain affected.
